### PR TITLE
Utilities: Some modernisation

### PR DESCRIPTION
**Pull request by AtkinsSJ**
Some DeprecatedString migration and a couple of other minor things.

(Previously also: removing a bunch of libc includes.)

### CODE REVIEW

File: Userland/Utilities/cksum.cpp [hunk 1]
```diff
@@ -4,26 +4,26 @@
  * SPDX-License-Identifier: BSD-2-Clause
```
Comment (krkk) [inline Userland/Utilities/cksum.cpp:4]: there's a typo in “cksum: Stop using DeprecatedString”:
> “**COn** the other hand[…]”

File: Userland/Utilities/stat.cpp [hunk 2]
```diff
@@ -10,10 +10,7 @@
 #include <LibCore/DateTime.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
-#include <grp.h>
-#include <pwd.h>
```
Comment (trflynn89) [inline Userland/Utilities/stat.cpp:14]: Same here, both of these are directly used, for example.

**Comment by AtkinsSJ** [comment]
Finally got around to modifying this. (Thanks for the update stalebot!)

- Dropped the "remove libc includes" commit.
- Fixed the typo krkk noticed.

**Comment by AtkinsSJ** [comment]
Rebased and fixed conflicts.

**Pull request merged**